solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapper.java

```diff
@@ -173,9 +177,26 @@ XMLMessage map(Message<?> message) {
 
 		if (setRawMessageHeader) builder.setHeader(SolaceMessageHeaderErrorMessageStrategy.SOLACE_RAW_MESSAGE, xmlMessage);
 
+		if (xmlMessage.getDestination() != null) {
+			builder.setHeaderIfAbsent(SolaceMessageHeaders.DESTINATION, xmlMessage.getDestination().getName());
```

Comment:
All messages send by scs will contain the destination already. But in case that you receive a messages send via mqtt from web browser or similar you will have this header not.

Reply:
This is already covered by #37 .  In particular its the by [the `solace_destination` header](https://github.com/SolaceDev/solace-spring-cloud/tree/staging/solace-spring-cloud-starters/solace-spring-cloud-stream-starter#solace-headers).

Reply:
Ok i will revert this point an rebase to #37 as soon as #37 made its way to master

Reply:
Ok i currently try to rebase agains the #37 but there i am unable to find any extraction of the solace_destination header. 
Can you please show me where you set it?

Reply:
The [SolaceHeaderMeta](https://github.com/SolaceDev/solace-spring-cloud/blob/staging/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/messaging/SolaceHeaderMeta.java) class abstracts away the getters and setters for the Solace message properties which might be why you didn't see it. [L17](https://github.com/SolaceDev/solace-spring-cloud/blob/staging/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/messaging/SolaceHeaderMeta.java#L17) is where the getter for the destination property is.

Then at [XMLMessageMapper.java#L189](https://github.com/SolaceDev/solace-spring-cloud/blob/staging/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapper.java#L189), this line calls `header.getValue().getReadAction().apply(xmlMessage)` which will call the previous function and set its return value to the `solace_destination` header.